File: our_dataset_preprocess.py
import os
from pydub import AudioSegment
import soundfile as sf

# Define the folder containing the .m4a files
input_folder = r'./our_chords_m4a/'
output_folder = r'./converted_chords_wav/'

# Create the output folder if it doesn't exist
if not os.path.exists(output_folder):
    os.makedirs(output_folder)

# Process each .m4a file in the input folder
for file_name in os.listdir(input_folder):
    if file_name.endswith('.m4a'):
        # Construct the full file path
        input_path = os.path.join(input_folder, file_name)
        
        # Load the audio file
        audio = AudioSegment.from_mp3(input_path)
        
        # Extract the first 2 seconds
        audio_2s = audio[:2000]
        
        # Resample to 44.1 kHz
        audio_2s = audio_2s.set_frame_rate(44100)
        
        # Export the audio as a .wav file (24-bit PCM)
        output_file_name = os.path.splitext(file_name)[0] + '.wav'
        output_path = os.path.join(output_folder, output_file_name)
        
        # Save as 24-bit PCM .wav
        audio_2s.export(output_path, format='wav', bitrate="24")

print("Conversion complete!")

File: preprocessing/our_dataset_preprocess.m
% Define the folder containing the .m4a files
inputFolder = 'our_chords_m4a';
outputFolder = 'our_chords_wav';

% Create the output folder if it doesn't exist
if ~exist(outputFolder, 'dir')
    mkdir(outputFolder);
end

% Get a list of all .m4a files in the input folder
m4aFiles = dir(fullfile(inputFolder, '*.m4a'));

% Process each .m4a file in the input folder
for i = 1:length(m4aFiles)
    % Get the full file path
    inputFilePath = fullfile(inputFolder, m4aFiles(i).name);
    
    % Read the audio file
    [audioData, originalSampleRate] = audioread(inputFilePath);
    
    % Extract the first 2 seconds
    durationInSeconds = 2;
    numSamples = min(durationInSeconds * originalSampleRate, size(audioData, 1));
    audioData_2s = audioData(1:numSamples, :);
    
    % Resample to 44.1 kHz if necessary
    targetSampleRate = 44100;
    if originalSampleRate ~= targetSampleRate
        audioData_2s = resample(audioData_2s, targetSampleRate, originalSampleRate);
    end
    
    % Construct the output file path
    [~, baseFileName, ~] = fileparts(m4aFiles(i).name);
    outputFileName = [baseFileName, '.wav'];
    outputFilePath = fullfile(outputFolder, outputFileName);
    
    % Save the audio as a 24-bit PCM .wav file
    audiowrite(outputFilePath, audioData_2s, targetSampleRate, 'BitsPerSample', 24);
end

disp('Conversion complete!');